### ISSUE TITLE
Detect Ansible 2.3-style warnings for Jinja2 in When

### ISSUE THREAD

**Issue by greg-hellings** [ansible/ansible-lint#261]
# Issue Type
- Feature request

# Ansible and Ansible Lint details

```
$ ansible --version
ansible 2.3.0.0
  config file = /home/ghelling/.ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jan 12 2017, 17:59:37) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
$ ansible-lint --version
ansible-lint 3.4.12
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

In Ansible 2.3, tasks are now issuing warnings when there is a "when" clause that includes Jinja2 templating text. These should be detected and alerted by ansible-lint as well. A PR is coming in a few minutes with this functionality added.

**Comment by ssbarnea** [comment]
Ability to lint jinja2 code would be a huge improvement on ansible-lint, maybe the best possible improvement. 

I know it is tricky but even some minimal linting support would be useful.

**Comment by ssbarnea** [comment]
Obsolete. Anyway, soon we will make use use ansible check functionality.
